### DISCUSSION TITLE
Categraf SNMP monitoring: why can't a numeric OID node be entered?

### DISCUSSION THREAD

**Discussion by pengjiangling** [ccfos/nightingale#1390] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/96814129/220511953-12ca51d0-5c62-4d6f-af40-33148701b6d6.png)
我的需求去监控ifNumber指标，通过mib手册查出oid为1.3.6.1.2.1.2.1，写入snmp配置发现categraf并没有采集数据
我用snmpwalk命令查出该oid的名称是IF-MIB::ifNumber.0
 snmpwalk -v 2c -c Tbkswitch 192.168.12.11 1.3.6.1.2.1.2.1
IF-MIB::ifNumber.0 = INTEGER: 56
然后更改配置，发现categraf好像只能识别这样的字符串。
可是MIB::name.索引号这种格式mib参考书里面并没有，难道需要我一个一个的用snmpwalk命令去替换解决问题么？

**Comment by UlricQin** [comment]
`--test` 测试结果截图

**Reply by pengjiangling** [reply]
![image](https://user-images.githubusercontent.com/96814129/220521769-a5cc766a-9a86-42db-800e-6d2bbc88ec81.png)
![image](https://user-images.githubusercontent.com/96814129/220521899-17248333-f4be-454e-a802-c8afe6b85777.png)

**Comment by pengjiangling** [comment]
还在吗你好 @UlricQin

**Reply by kongfei605** [reply]
issue跟mail一样，区别于即时聊天，又能沉淀积累，这个问题我来跟进。

**Reply by pengjiangling** [reply]
好的，能否告知一下期限，我方便处理对应的解决方案

**Reply by pengjiangling** [reply]
@kongfei605

**Reply by kongfei605** [reply]
另外，采集的指标名称是snmp_配置中的name，不是oid

**Comment by kongfei605** [comment, marked as answer]
oid后面少个零 ，我这边测试是可以的。

<img width="1075" alt="image" src="https://user-images.githubusercontent.com/12164762/220899314-22d2543a-ba2a-4942-8148-9914a9b0e327.png">

**Comment by kongfei605** [comment]
`snmptranslate -Td RFC1213-MIB::ifNumber.0`

<img width="966" alt="image" src="https://user-images.githubusercontent.com/12164762/220898996-c6b90daa-a9cd-4690-b67f-c2ae380aeaa8.png">

`snmpget -v2c -c xxxx 172.30.x.x 1.3.6.1.2.1.2.1`
会提示
```
IF-MIB::ifNumber = No Such Instance currently exists at this OID
```

<img width="971" alt="image" src="https://user-images.githubusercontent.com/12164762/220904230-de381009-836c-451d-9cc4-904be07b1f48.png">

**Reply by pengjiangling** [reply]
![image](https://user-images.githubusercontent.com/96814129/221067104-4883febb-973b-4729-b2e8-305e8587104a.png)
我这边没有提示。

**Reply by kongfei605** [reply]
配置中加上.0  然后--test --debug  看看输出

**Reply by kongfei605** [reply]
snmptranslate 带0 和不带0 的输出也看看

**Reply by pengjiangling** [reply]
配置中oid加上.0，能够获取到数据，不拿点0，无法获取
snmptranslate
![image](https://user-images.githubusercontent.com/96814129/221074382-cedaa80c-7f98-4172-bc72-c9148147345c.png)

**Reply by kongfei605** [reply]
你看你translate 不带0， 应该是1.3.6.1.2.1.1.5 （最后一行）。
这个插件是通过snmptranslate转换，然后snmpget/walk的。全局的table（可能这个说法不太精确）是snmptranslate 然后snmpget，所以如果translate有问题，就会获取不到了。
不太了解你的设备特性，我这边如果oid不存在 会直接报错。 我这边也优化了插件的出错提示，如果oid对应的设备找不到会直接报错了。可以在你的环境里再试试。
<img width="1543" alt="image" src="https://user-images.githubusercontent.com/12164762/221087756-80990f50-82af-448c-aaaf-6795300cc660.png">

**Reply by pengjiangling** [reply]
是最新版么

**Reply by kongfei605** [reply]
发布了一个版本，v0.2.31
